pkg/metainfo/kvmetainfo/objects_test.go

```diff
@@ -83,43 +99,77 @@ func TestGetObjectStream(t *testing.T) {
 		_, err = db.GetObjectStream(ctx, bucket.Name, "")
 		assert.True(t, storage.ErrEmptyKey.Has(err))
 
-		_, err = db.GetObjectStream(ctx, "non-existing-bucket", "test-file")
+		_, err = db.GetObjectStream(ctx, "non-existing-bucket", "small-file")
 		assert.True(t, storj.ErrBucketNotFound.Has(err))
 
 		_, err = db.GetObject(ctx, bucket.Name, "non-existing-file")
 		assert.True(t, storj.ErrObjectNotFound.Has(err))
 
 		stream, err := db.GetObjectStream(ctx, bucket.Name, "empty-file")
 		if assert.NoError(t, err) {
-			assertStream(ctx, t, stream, "empty-file", []byte{})
+			assertStream(ctx, t, stream, "empty-file", 0, []byte{})
 		}
 
-		stream, err = db.GetObjectStream(ctx, bucket.Name, "test-file")
+		stream, err = db.GetObjectStream(ctx, bucket.Name, "small-file")
 		if assert.NoError(t, err) {
-			assertStream(ctx, t, stream, "test-file", []byte("test"))
+			assertStream(ctx, t, stream, "small-file", 4, []byte("test"))
+		}
+
+		stream, err = db.GetObjectStream(ctx, bucket.Name, "large-file")
+		if assert.NoError(t, err) {
+			assertStream(ctx, t, stream, "large-file", int64(32*memory.KB), data)
 		}
 	})
 }
 
-func assertStream(ctx context.Context, t *testing.T, stream storj.ReadOnlyStream, path storj.Path, content []byte) bool {
+func assertStream(ctx context.Context, t *testing.T, stream storj.ReadOnlyStream, path storj.Path, size int64, content []byte) {
 	assert.Equal(t, path, stream.Info().Path)
 
 	segments, more, err := stream.Segments(ctx, 0, 0)
 	if !assert.NoError(t, err) {
-		return false
+		return
 	}
 
 	assert.False(t, more)
 	if !assert.Equal(t, 1, len(segments)) {
```

Comment:
nit -- you can do assert.NotEqual(...

Reply:
Here is similar. I want to assert that there is only one segment. But if the assertion fails, I want to interrupt the test to avoid the next assertions, because they may panic.